J/JSBSim/build_tarballs.jl

```diff
@@ -0,0 +1,58 @@
+using BinaryBuilder
+
+name = "JSBSim"
+version = v"1.1.6"
+julia_version = v"1.5.3"
+
+# Collection of sources required to build JSBSim
+sources = [
+    GitSource("https://github.com/bcoconni/jsbsim.git",
+              "d699c3bcd90bf9ce2a64c1fe3091fcf39714ccfd"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/jsbsim
+mkdir build && cd build
+FLAGS=()
+if [[ "${target}" == *-mingw* ]]; then
+    FLAGS+=(-DCMAKE_CXX_FLAGS_RELEASE="-D_POSIX_C_SOURCE")
+fi
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_FIND_ROOT_PATH=${prefix} \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DBUILD_DOCS=OFF \
+    -DBUILD_PYTHON_MODULE=OFF \
+    -DBUILD_JULIA_PACKAGE=ON \
+    -DJulia_PREFIX=${prefix} \
+    "${FLAGS[@]}" \
+    ..
+cmake --build . --target JSBSimJL -- -j${nproc}
+install_license $WORKSPACE/srcdir/jsbsim/COPYING
+cp julia/*JSBSimJL*.$dlext $libdir/.
+cp ../julia/JSBSim.jl $prefix/.
+"""
+
+ARGS
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+include("../../L/libjulia/common.jl")
+platforms = libjulia_platforms(julia_version)
+platforms = expand_cxxstring_abis(supported_platforms())
+
+filter!(p -> libc(p) != "musl", platforms) # muslc is not supported
```

Comment:
Just out of curiosity: why / how does that manifest? Naively, I'd expect code that manages to work on Linux with glibc and macOS *and* mingw to also work with musl. Just wondering what the breakage is?

Reply:
Because building the package with musl fails miserably. I have commented out the line so that you can see by yourself.

Reply:
All I saw so far are basic portability issues that should be not too difficult to resolve. E.g. `src/input_output/FGfdmSocket.cpp` is missing `#include <sys/select.h` which is required for using `select`, but many C libraries are lax about that and their headers end up including each other in seemingly arbitrary ways, which can occlude such issues, until you hit a system with a different C library... musl tries hard to avoid this, which to me is one of the best things about it: it forces you to write code that actually follows the definitions given in the C / POSIX standard, so if you get your code to work with musl, then it already is on a good path towards being properly portable.

Similar in `src/simgear/misc/strutils.cxx`: it's check for the signature of `strerror_r` relies on preprocessor checks (inferior to properly detecting it with a `configure` script, but of course much quicker to write down), and that logic is flawed; it equates `_GNU_SOURCE` being defined as "this is glibc", which is not at all what `_GNU_SOURCE` means. Instead, it probably should check for `__GLIBC__`.

Reply:
Actually the `strerror_r` is even more involved, as it not just depends on whether GLIBC is used, but indeed also on #defines like `_GNU_SOURCE`. From the glibc man page on `strerror_r`:
```
   Feature Test Macro Requirements for glibc (see feature_test_macros(7)):

       strerror_r():
           The XSI-compliant version is provided if:
           (_POSIX_C_SOURCE >= 200112L) && !  _GNU_SOURCE
           Otherwise, the GNU-specific version is provided.
```

Matching this correctly to cover all cases is tricky, but I think overall changing this near the end of https://github.com/bcoconni/jsbsim/blob/master/src/simgear/misc/strutils.cxx
```
#elif defined(_GNU_SOURCE)
  return std::string(strerror_r(errnum, buf, sizeof(buf)));
```
to something like this
```
#elif defined(__GLIBC__ ) && defined(_GNU_SOURCE)
  return std::string(strerror_r(errnum, buf, sizeof(buf)));
```
should do the trick, at least for the purposes of this PR

Reply:
Thanks for the feedback. I'll investigate that.